ui/raidboss/data/07-dt/hunts/shaaloani.ts

```diff
@@ -434,6 +562,233 @@ const triggerSet: TriggerSet<Data> = {
+      id: 'Hunt Ttokrrone Sandspout',
+      type: 'StartsUsing',
+      netRegex: { id: ['91C1', '91C2', '91C3', '91C4'], source: 'Ttokrrone', capture: true },
+      durationSeconds: 4.9,
+      response: (data, matches, output) => {
+        const pattern = data.ttokSandOrbPatterns[data.ttokSandOrbOnSet];
+        let sandspoutPattern = TtokSafeSpots.Out;
+        let awaySide = 'front';
+        let rotation = 0;
+        let dir1: TtokrroneDodge = 'unknown';
+        let dir2: TtokrroneDodge = 'unknown';
+        // cactbot-builtin-response
+        output.responseOutputStrings = ttokrroneTempestSandspoutOutputStrings;
+
+        if (matches.id === '91C1') { // front cleave
+          sandspoutPattern = 0b101_000_000_111; // except in front
+          awaySide = output.front!();
+          rotation = 0;
+        } else if (matches.id === '91C2') { // back cleave
+          sandspoutPattern = 0b111_000_000_101; // except behind
+          awaySide = output.rear!();
+          rotation = 2;
+        } else if (matches.id === '91C3') { // right cleave
+          awaySide = output.rightFlank!();
+          rotation = 1;
+        } else if (matches.id === '91C4') { // left cleave
+          awaySide = output.leftFlank!();
+          rotation = 3;
+        }
+
+        if (pattern) {
+          const safeSpot = sandspoutPattern & pattern;
+          let backFrontSpot: TtokSafeSpots = 0b111_111_111_111;
+          if (safeSpot & TtokSafeSpots.Back) {
+            backFrontSpot = TtokSafeSpots.Back;
+            dir1 = 'back';
+          } else if (safeSpot & TtokSafeSpots.Front) {
+            backFrontSpot = TtokSafeSpots.Front;
+            dir1 = 'front';
+          }
+
+          if (safeSpot & backFrontSpot & TtokSafeSpots.Right) {
+            dir2 = 'right';
+          } else if (safeSpot & backFrontSpot & TtokSafeSpots.Left) {
+            dir2 = 'left';
+          }
+          data.ttokSandOrbOnSet++;
+          data.ttokRotated = rotation;
+          return {
+            alertText: output.triple!({
+              inOut: output.outOfHitbox!(),
+              dir2: output[dir1]!(),
+              dir3: output[dir2]!(),
+            }),
+          };
+        }
+        return {
+          infoText: output.awayFrom!({ out: output.outOfHitbox!(), dir: awaySide }),
+        };
+      },
+    },
+    // The boss does either in; out; in-right + out-left; or out-right + in-left
+    // can be combined with sand orbs explosions.
+    {
+      id: 'Hunt Ttokrrone Desert Tempest',
+      type: 'StartsUsing',
+      netRegex: { id: ttokrroneDesertTempestIds, source: 'Ttokrrone', capture: true },
+      durationSeconds: 7,
+      alertText: (data, matches, output) => {
+        const tempest = ttokrroneDesertTempest[matches.id]!;
+        const pattern = data.ttokSandOrbPatterns[data.ttokSandOrbOnSet];
+        let inOut: TtokrroneDodge = 'unknown';
+        let dir2: TtokrroneDodge | null = null;
+        let dir3: TtokrroneDodge | null = null;
```

Comment:
```suggestion
        let rightLeft: TtokrroneDodge | null = null;
        let backFront: TtokrroneDodge | null = null;
```

Might help readability a little to use more distinctive naming.